File: DTC.Api/Dtos/Match/MatchDto.cs
using DTC.Api.Dtos.Board;
using DTC.Api.Dtos.MatchParticipant;
using DTC.Api.Enums;
using DTC.Api.Models;
using System.Security.Cryptography.Pkcs;

namespace DTC.Api.Dtos.Match
{
    public class MatchDto
    {
        public int Id { get; set; }
        public int RoundId { get; set; }
        public int? GroupId { get; set; }
        public BoardDto? Board { get; set; }

        public MatchStatus Status { get; set; }
        public DateTimeOffset? PlannedStart { get; set; }
        public DateTimeOffset? PlannedEnd { get; set; }
        public DateTimeOffset? ActualStart { get; set; }
        public DateTimeOffset? ActualEnd { get; set; }

        public List<MatchParticipantDto> Participants { get; set; } = new();
    }
}

File: dtc.frontend/src/dtos/match/MatchDto.ts
import z from "zod";
import { MatchStatus } from "../../enums/MatchStatus";
import { matchParticipantDtoSchema } from "../matchParticipant/MatchParticipantDto";

export const matchDtoSchema = z.object({
  id: z.number(),
  roundId: z.number(),
  groupId: z.number().int().nullable(),
  boardId: z.number().int().nullable(),
  status: z.enum(MatchStatus),
  plannedStart: z.iso.datetime({ offset: true }).nullable(),
  plannedEnd: z.iso.datetime({ offset: true }).nullable(),
  actualStart: z.iso.datetime({ offset: true }).nullable(),
  actualEnd: z.iso.datetime({ offset: true }).nullable(),
  participants: z.array(matchParticipantDtoSchema),
});

export type MatchDto = z.infer<typeof matchDtoSchema>;
